### ISSUE TITLE
Architecture issue: Organising decorators.

### ISSUE THREAD

**Issue by nil0x42** [nil0x42/phpsploit#20]
Currently, decorators are organised this way:

``` txt
src/decorators/
├── __init__.py
├── isolate_io_context.py
└── readonly_settings.py
```

I love this way, so decorators can be used in the following manners:

``` python
import decorators
@decorators.isolate_io_context
def foo():
    pass
```

or

``` python
from decorators import isolate_io_context
@isolate_io_context
def foo():
    pass
```

Therefore, i got a cyclic dependency tree, while trying to use a decorator at an initialization layer,
because the other decorator was loaded too while importing `decorators`.

For the moment, i did a local copy of isolate_io_context on src/ui/input/expect.py, which is they
file where i found that issue.

Is there a way to use decorators import in a clean way, without loading all of them when just importing `decorators` ?
I'd like a way where a file is really loaded only when callding it (e.g: import decorator.isolate_io_context).

I think something like this:

``` python
import decorators.iolate_io_context
@decorators.isolate_io_context.isolate_io_context
def foo():
    pass
```

is ugly...

I m open to any suggestion..

**Comment by nil0x42** [comment]
Thanks for the feedback,
I'll consider doing this.
(soon if possible, as ever ^^)

**Comment by nil0x42** [comment]
fixed by 965657558e5b55c24f8f07c6ac6241ed0750dfa
